Hard/1. Shopping Cart Problem/solutions/damini31_DaminiChauhan/Hard_ShoppingCart.py

```diff
@@ -0,0 +1,618 @@
+                  while(j!=0):
+                      p_check=True
+                      while(p_check):
+                         t=input("\nEnter the Product ID: ") 
+                         if(len(t)==0):
+                             print("\nMandatory Field.Please enter Product ID.")
+                             p_check = True
+                         else:
+                             try:
+                                t=int(t)
+                                y=is_present(pf['Product ID '],t)
+                                
+                                if(y == True):
+                                    p_list.append(t)
+                                    q_check=True
+                                    while(q_check):
+                                       r=input("\nEnter the quantity of the product: ")
+                                       if(len(r)==0 or r == "0"):
+                                            print("\nMandatory Field.Quantity Cannot be zero or blank.")
+                                            q_check = True
+                                       else:
+                                           try:
+                                               r=int(r)
+                                               q_list.append(r)
+                                               j=int(input("\nDo you want to add more items?Press 0 to exit and 1 to add more items?  "))
+                                               if(j==0):
+                                                   check=False
+                                               else:
+                                                   check=True 
+                                               q_check=False
+                                               p_check = False
+                                            
+                                           except(ValueError):
+                                                print("\nQuantity is numeric.Please Re-enter ")
+                                                q_check=True
+                                       
+                                       
+          
+                                
+                                else:
+                                    print("\nEnter the correct Product ID")
+                                    p_check = True
+                            
+                             except(ValueError):
+                                print("\nProduct ID is numeric.Please Re-enter")
+                                p_check=True
+                                
+                delivery=int(input("\nPress 0 for Home Delivery and 1 for Takeaway:  "))
+                if(delivery==0):
+                  distance=int(input("What is the distance between your residence and the outlet: "))
+                  d_check=True
+                  while(d_check):
+                     distance=input("What is the distance between your residence and the outlet: ")
```

Comment:
It is asking to enter distance twice now when I select home delivery.
![Screenshot 2020-07-31 at 1 55 03 PM](https://user-images.githubusercontent.com/54791364/89015909-7b162e00-d335-11ea-99a1-06288616ee9a.png)